workflow/controller/operator.go

```diff
@@ -1676,6 +1685,29 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 	return node, nil
 }
 
+//Checks the template if it exceeds its deadline
+func (woc *wfOperationCtx) checkTemplateDeadline(tmpl *wfv1.Template, node *wfv1.NodeStatus) (*time.Time, error) {
+	woc.log.Debugf("Check Template deadline")
+	if node == nil {
+		return nil, nil
+	}
+
+	if tmpl.TimeoutDuration != nil {
+		tmplMaxDuration, err := parseStringToDuration(tmpl.TimeoutDuration.StrVal)
```

Comment:
String()

Reply:
Should this be `.String()` like @alexec mentioned?